azurerm/internal/services/notificationhub/notification_hub_resource.go

```diff
@@ -231,33 +232,30 @@ func resourceArmNotificationHubRead(d *schema.ResourceData, meta interface{}) er
 	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	id, err := azure.ParseAzureResourceID(d.Id())
+	id, err := parse.NotificationHubID(d.Id())
```

Comment:
can we add this function to the Import check too, which should raise this at that point ([example](https://github.com/terraform-providers/terraform-provider-azurerm/blob/5785c873ab5d13043b91a71d5904d9612af338c0/azurerm/internal/services/analysisservices/analysis_services_server_resource.go#L40-L43))